packages/fast-usdc/src/clientSupport.js

```diff
@@ -0,0 +1,110 @@
+// @ts-check
+import { AmountMath } from '@agoric/ertp';
+import { assertAllDefined } from '@agoric/internal';
+
+// Reusable amount scaling similar to inter-protocol's approach
+const COSMOS_UNIT = 1_000_000n;
+const scaleDecimals = num => BigInt(num * Number(COSMOS_UNIT));
+
+/**
+ * @param {Pick<
+ *   import('@agoric/vats/tools/board-utils.js').AgoricNamesRemotes,
+ *   'brand'
+ * >} agoricNames
+ * @param {Instance} instance
+ * @param {object} opts
+ * @param {string} opts.offerId
+ * @param {number} opts.amount
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+const makeDepositOffer = ({ brand }, instance, { offerId, amount }) => {
```

Comment:
`want` is required, and the keyword is `USDC`, not `Deposit`.

I'd like us to re-use the existing fast usdc deposit type as in:

https://github.com/Agoric/agoric-sdk/blob/f0ccde6550d7f0d22fd489205cb12a6598faa2c9/packages/fast-usdc/src/cli/lp-commands.js#L104-L112

We could calculate the want for them if they pass in `metrics` or `shareWorth` as in...

https://github.com/Agoric/agoric-sdk/blob/f0ccde6550d7f0d22fd489205cb12a6598faa2c9/packages/fast-usdc/src/cli/lp-commands.js#L101-L102

We might want a `slippage` parameter.